python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/traffic-policies.j2

```diff
@@ -55,19 +55,22 @@ traffic-policies
 {%     for policy in traffic_policies.policies | arista.avd.natural_sort('name') %}
    !
    traffic-policy {{ policy.name }}
-{%         if policy.matches is arista.avd.defined %}
 {# COUNTER CREATION SECTION #}
-{%             set transient_values = namespace() %}
-{%             set transient_values.counters = [] %}
-{%             for match in policy.matches %}
-{%                 if match.actions.count is arista.avd.defined %}
-{%                     do transient_values.counters.append(match.actions.count | string) %}
-{%                 endif %}
-{%             endfor %}
-{%             if transient_values.counters | length > 0 %}
-      counter {{ transient_values.counters | arista.avd.natural_sort | join(' ') }}
-      !
+{%         set transient_values = namespace(counters=[]) %}
+{%         if policy.counters is arista.avd.defined %}
+{%             do transient_values.counters.extend(policy.counters) %}
+{%         endif %}
+{# TODO: Remove this logic for counter creation in AVD 6.0, only use the `policy.counters` list #}
+{%         for match in policy.matches | arista.avd.default([]) %}
+{%             if match.actions.count is arista.avd.defined %}
+{%                 do transient_values.counters.append(match.actions.count) %}
 {%             endif %}
+{%         endfor %}
```

Comment:
I understand that we are going to remove this but still why aren't we writing this logic in else block?